src/main/java/org/aksw/simba/squirrel/deduplication/hashing/impl/IntervalBasedMinHashFunction.java

```diff
@@ -0,0 +1,54 @@
+package org.aksw.simba.squirrel.deduplication.hashing.impl;
+
+import org.aksw.simba.squirrel.deduplication.hashing.HashValue;
+import org.aksw.simba.squirrel.deduplication.hashing.RDFHashFunction;
+import org.apache.jena.graph.NodeFactory;
+import org.apache.jena.graph.Triple;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+/**
+ * Use Min hash algorithm to compute hash values for triples.
+ */
+public class IntervalBasedMinHashFunction implements RDFHashFunction {
+
+    /**
+     * The number n for the 2^n Intervals for the Hashing.
+     */
+    private int powerNumberOfIntervals;
+
+    /**
+     * Constructor.
+     *
+     * @param powerNumberOfIntervals The powered number of Intervalls for the hashing.
+     */
+    public IntervalBasedMinHashFunction(int powerNumberOfIntervals) {
+        this.powerNumberOfIntervals = powerNumberOfIntervals;
+    }
+
+    @Override
+    public HashValue hash(List<Triple> triples) {
+        List<Integer> listHashValues = new ArrayList<>(powerNumberOfIntervals);
+        for (int i = 0; i < powerNumberOfIntervals; i++) {
+            listHashValues.add(i, null);
+        }
+        int intervalSize = 2 * Integer.MAX_VALUE / (int) Math.pow(2, powerNumberOfIntervals);
+        for (Triple triple : triples) {
+            Integer hash = triple.hashCode();
+            int currentInterval = Integer.MIN_VALUE;
+            for (int i = 0; i < Math.pow(2, powerNumberOfIntervals); i++) {
```

Comment:
1. this for loop asks in _every iteration_ for `Math.pow(2, powerNumberOfIntervals)` and typically, `Math.pow` is expensive. You already know how many intervals you have so you don't have to calculate this number again and again :wink: 
2. Yo simply don't need this loop. With a given number, you can directly calculate the cell with which you have to compare it. Yes, with bit shifting magic :wink: Please think about it.